proto/bootz.proto

```diff
@@ -0,0 +1,167 @@
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+
+import "github.com/openconfig/gnsi/authz/authz.proto";
+import "github.com/openconfig/gnsi/certz/certz.proto";
+import "github.com/openconfig/gnsi/credentialz/credentialz.proto";
+import "github.com/openconfig/gnsi/pathz/pathz.proto";
+
+
+option go_package = "github.com/openconfig/bootz/proto/bootz";
+
+
+// The Bootz service definition.
+// Bootz service is hosted by the bootz-server, and the device is the
+// client (as opposed to most g*i services, which are hosted by the device).
+service Bootstrap {
+  // This RPC returns the data required to put the device in a known state
+  // (i.e. OS, bootloader password, etc) and applies an initial device
+  // configuration.
+  rpc GetBootstrapData(GetBootstrapDataRequest) returns (GetBootstrapDataResponse) {}
+   
+  // This RPC is used by the device to indicate successful application of
+  // BootstrappingData. The Bootz system will proceed to the next step if
+  // a SUCCESS is reported, otherwise it will retry or put the device in an
+  // out-of-service state. The device should validate the server's identity
+  // against the server_trust_cert it obtained in GetBootstrappingDataResponse.
+  rpc ReportStatus(ReportStatusRequest) returns (EmptyResponse) {}
+}
+
+// Initial request from device to the bootz-server. Contains a detailed
+// description of the chassis and its constituent control cards.
+message GetBootstrapDataRequest {
+  ChassisDescriptor chassis_descriptor = 1;
+  ControlCardState control_card_state = 2;
+  // A nonce that the bootstrap server should use when signing the response.
+  string nonce = 1001;
+}
+
+// Structured message containing the chassis details, and listing its component
+// control-card modules.
+message ChassisDescriptor {
+  string manufacturer = 1;
+  string part_number = 2;
+  // This field is populated for fixed form-factor chassis, and empty for
+  // modular chassis.
+  string serial_number = 3;
+  // For modular chassis this field will be used to list the control cards
+  // to be bootstrapped.
+  repeated ControlCard control_cards = 4;
+}
+
+// Details of the control card, including serial-number and the location
+// of the card on the chassis.
+message ControlCard {
+  string part_number = 1;
+  string serial_number = 2;
+  int32 slot = 3;
+}
+
+// The status of each control card (i.e. initialized or not). For first install
+// of the chassis, both control cards are in NOT_INSTALLED state. For 
+// install a hot-swapped standby, the status of the active card is INSTALLED, 
```

Comment:
nit: Is ACTIVE a better word than INSTALLED?

Reply:
i think that conflates different things
this is answering is the card installed - it doesn't imply which card is active/standby or any thing about the operational state of the card

Reply:
I think the text difficult to understand because 'INSTALLED' is not clearly defined.  Does install mean the physical card is plugged in?  Or does it mean an operating system is installed?  Typically devices (outside of the whitebox ecosystem)  with factory defaults do include an operating system which is "installed".  

After reading this a few times, it seems like installed might be referring to the state of the bootstrap process.  ie: an indication of whether the bootstrap process has completed successfully or not.  

I think a clarification is needed

Reply:
// The status of each control card (i.e. initialized or not). For first install
// of the chassis, both control cards are in NOT_INSTALLED state. For 
// install a hot-swapped standby, the status of the active card is INSTALLED,
// and that of the standby is NOT_INSTALLED. For fixed form-factor chassis,
// the serial number is that of the chassis, and status is NOT_INSTALLED.


Suggested reword of the entire comment (lines 77-81).
// The status of each control card (bootstrapped or not). At the 
// beginning of the bootstrap process (i.e. before a ReportStatus
// has completed with BOOTSTRAP_STATUS_SUCCESS), both 
// control cards are in NOT_INSTALLED state. 
// When a control-card is hot-swapped, the bootz process is
// performed with the swapped control-card having a status of
// NOT_INSTALLED, and the other card in INSTALLED state.

Reply:
Building off your suggestion below.  I also changed the enum to use NOT_INITIALIZED and INITIALIZED to be consistent with the readme which uses these terms.

If an `INITIALIZED` control card is removed and reinserted, should it still be in `INITIALIZED` state?  Or should it be `NOT_INITIALIZED`?  I would expect it to remain `INITALIZED`, yes?

// ControlCardState indicates if the bootstrap process has 
// successfully completed.
//
// At the beginning of the bootstrap process (i.e. before a 
// ReportStatus has completed with BOOTSTRAP_STATUS_SUCCESS),
// all control cards (1 or 2) are in NOT_INITIALIZED state.
// If bootstrap successfully completes for a control-card, the 
// ControlCardStatus changes to INITIALIZED.
//
// Once a control-card is in INITIALIZED state, it may remain in that
// state indefinitely, even if removed and reinstalled in a chassis.  
// 
// When there are 2 control-cards present and INITIALIZED and 1 is 
// removed, the remaining control card remains in INITIALIZED state.
// If a new control-card is inserted which has never completed the
// bootstrap process, it will start with NOT_INITIALIZED state.
message ControlCardState {
  string serial_number = 1;
  ControlCardStatus status = 2;
}

enum ControlCardStatus {
  CONTROL_CARD_STATUS_UNSPECIFIED = 0;       // the bootstrap process status is not reported
  CONTROL_CARD_STATUS_ NOT_INITIALIZED = 1;  //  the bootstrap process has not successfully completed
  CONTROL_CARD_STATUS_ INITIALIZED = 2;            //  the bootstrap process has successfully completed
}